src/pages/index.js

```diff
@@ -1,275 +1,242 @@
-const currentProfileName = document.querySelector('.profile__name');
-const currentProfileDescription = document.querySelector('.profile__description');
-//#endregion
+const cardSection = new Section({
+  containerSelector: `.cards__list`,
+  createItem: (data) => {
+    return makeNewCard(data);
+  },
+  initialize: (route) => {
+    dbApi.getBatchData(route).then((data) => {
+      cardSection.createItems(data);
+      cardSection.renderItems();
+    });
+  },
+});
 
-//#region new post modal declarations
-const defaultLinkField = 'Paste a link to the picture';
-const defaultCaptionField = 'Type your caption'
+cardSection.initialize(routeData.getCards);
+
+const userProfile = new UserProfile({
+  initialize: () => {
+    dbApi
+      .request(routeData.getUserInfo)
+      .then((data) => {
+        userProfile.setProfileData({
+          userName: data.name,
+          profileDescription: data.about,
+          imageUrl: data.avatar,
+        });
+        editProfileForm.setDefaultInputs(userProfile.getUserProfile());
+      })
+      .catch((err) => {
+        console.error(`Failed to get user data: ${err}`);
+      });
+  },
+});
 
-const postModal = document.querySelector('#post-modal');
-const postLink = postModal.querySelector('#link');
-const postCaption = postModal.querySelector('#caption');
-postModal.addEventListener('click',checkForModalOverlayClick);
-document.querySelector('.profile__add-button').addEventListener('click',onNewPostClick);
-document.querySelector('#new-post-modal__close').addEventListener('click',onCloseModalClick);
-const cardForm = document.forms['new-post-modal__form']
-cardForm.addEventListener('submit',onPostModalSubmit);
-//#endregion
+userProfile.initialize();
+
+const editAvatarForm = new ModalWithForm({
+  modalSelector: modalSelectors.editAvatarSelector,
+  onSubmitCallback: (inputValues) => {
+    return new Promise((resolve, reject) => {
+      dbApi
+        .request(routeData.updateUserAvatar, {
+          avatar: inputValues["avatar-link"],
+        })
+        .then(() => {
+          userProfile.setUserImage(inputValues["avatar-link"]);
+          resolve();
+        })
+        .catch(() => reject());
+    });
+  },
+});
 
-//#region preview modal declarations
-const previewModal = document.querySelector('#preview-modal');
-const previewImage = document.querySelector('.modal__preview-image');
-const previewTitle = document.querySelector('#preview-title');
-previewModal.addEventListener('click',checkForModalOverlayClick);
-document.querySelector('#preview-modal__close').addEventListener('click',onCloseModalClick);
-//#endregion
+const editAvatarFormValidator = new FormValidator(
+  validationSettings,
+  editAvatarForm.getFormElement()
+);
+editAvatarFormValidator.enableValidation();
+
+const editProfileForm = new ModalWithForm({
+  modalSelector: modalSelectors.editProfileModal,
+  onSubmitCallback: (inputValues) => {
+    return new Promise((resolve, reject) => {
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```

Reply:
That is super cool. Thank you! Excited to try and implement this.